tasks-standard/src/main/scala/sbt/std/Streams.scala

```diff
@@ -155,13 +157,30 @@ object Streams {
 
       def log(sid: String): ManagedLogger = mkLogger(a, text(sid))
 
-      def make[T <: Closeable](a: Key, sid: String)(f: File => T): T = synchronized {
+      // Creates log file before returning.
+      // Due to task concurrency this call can fail at any point in time.
+      // A long-term fix might be split log directory, use native log4j appender,
+      // and/or block `clean` task until it can have exclusive access to `target` directory.
+      def make[T <: Closeable](a: Key, sid: String)(f: File => T): T = {
         checkOpen()
         val file = taskDirectory(a) / sid
-        IO.touch(file, false)
-        val t = f(file)
-        opened ::= t
-        t
+        def doMake: T = synchronized {
+          IO.touch(file, false)
+          val t = f(file)
+          opened ::= t
+          t
+        }
+        @tailrec def retry(n: Int)(run: => T): T =
+          catching(classOf[IOException]).either(run) match {
+            case Right(x) => x
+            case Left(e) =>
+              if (n <= 0) throw e
+              else {
+                Thread.sleep(100)
+                retry(n - 1)(run)
+              }
+          }
+        retry(30) { doMake }
```

Comment:
Basically I wasn't sure if `clean` is the only culprit of concurrency issue. But we can introduce one parameter at a time, and take this commit out and see how it goes.